src/ui/public/courier/data_source/search_source.js

```diff
@@ -288,5 +292,16 @@ export function SearchSourceProvider(Promise, Private, config) {
     }
   };
 
+  SearchSource.prototype.clone = function () {
+    const clone = new SearchSource(this.toString());
+    clone.set('index', this.get('index'));
```

Comment:
Ah right, I forgot saved_object.js handles "hydrating" the index pattern on the search source, for whatever reason.

This seems like a reasonable enough workaround to me, would you mind just adding a comment explaining what it's for?